pkg/ipamd/datastore/data_store.go

```diff
@@ -144,18 +146,24 @@ type ENI struct {
 	IsTrunk bool
 	// IsEFA indicates whether this ENI is tagged as an EFA
 	IsEFA bool
+	// IsPDenabled indicates whether prefix delegation is enabled
+	IsPDenabled bool
 	// DeviceNumber is the device number of ENI (0 means the primary ENI)
 	DeviceNumber int
 	// IPv4Addresses shows whether each address is assigned, the key is IP address, which must
 	// be in dot-decimal notation with no leading zeros and no whitespace(eg: "10.1.0.253")
 	IPv4Addresses map[string]*AddressInfo
+	// Key is the prefix and value is prefix, used IPs under the prefix
+	IPv4Prefixes map[string]*ENIPrefix
 }
 
 // AddressInfo contains information about an IP, Exported fields will be marshaled for introspection.
 type AddressInfo struct {
 	IPAMKey        IPAMKey
 	Address        string
 	UnassignedTime time.Time
+	Prefix         string
+	IPIndex        int
```

Comment:
I think we can derive `IPIndex` on the fly from `Address` and `Prefix` - even better if we store `Address` and `Prefix` as parsed `net.IP` (or `net.IPNet`) types.

Reply:
Fixed this to be net.IP